src/main/kotlin/mate/academy/HelloWorldAnonymousClasses.kt

```diff
@@ -8,6 +8,40 @@ class HelloWorldAnonymousClasses {
     }
 
     fun sayHello(names: List<String>): List<String> {
-        // TODO: implement
+        val englishGreeting = object : HelloWorldGreeting {
+            override fun greet(): String {
+                return "Hello world"
+            }
+
+            override fun greetSomeone(someone: String): String {
+                return "Hello $someone"
+            }
+        }
+
+        val frenchGreeting = object : HelloWorldGreeting {
+            override fun greet(): String {
+                return "Salut tout le monde"
+            }
+
+            override fun greetSomeone(someone: String): String {
+                return "Salut $someone"
+            }
+        }
+
+        val spanishGreeting = object : HelloWorldGreeting {
+            override fun greet(): String {
+                return "Hola, mundo"
+            }
+
+            override fun greetSomeone(someone: String): String {
+                return "Hola, $someone"
+            }
+        }
+        val greeters = listOf(englishGreeting, frenchGreeting, spanishGreeting)
+        val generalGreetings = greeters.map { it.greet() }
+        val personalizedGreetings = names.flatMap { name ->
```

Comment:
Here may be minor update via check if names list is not empty. 
Not mistake just note